frontend/app/brains-management/[brainId]/components/BrainManagementTabs/components/SettingsTab/components/PrivateAccessConfirmationModal/hooks/useAccessConfirmationModal.ts

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from "react";
+import { UseFormResetField } from "react-hook-form";
+
+import { BrainConfig, BrainStatus } from "@/lib/types/brainConfig";
+
+type UseAccessConfirmationModalProps = {
+  status: BrainStatus;
+  setValue: (name: "status", value: "private" | "public") => void;
+  isStatusDirty: boolean;
+  resetField: UseFormResetField<BrainConfig>;
+};
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const useAccessConfirmationModal = ({
+  status,
+  isStatusDirty,
+  resetField,
+}: UseAccessConfirmationModalProps) => {
+  const [isAccessModalOpened, setIsAccessModalOpened] = useState(false);
+
+  useEffect(() => {
+    console.log({
+      status,
+      isStatusDirty,
+    });
+    if (!isStatusDirty) {
+      return;
+    }
+    setIsAccessModalOpened(true);
+  }, [status]);
+
+  const closeModal = () => {
+    setIsAccessModalOpened(false);
+  };
+
+  const onCancel = () => {
+    closeModal();
+    // setValue("status", "public");
+    resetField("status", {
+      defaultValue: status === "private" ? "public" : "private",
```

Comment:
Could we simplify this with:
```suggestion
      defaultValue: status
```
(since `status` is already either `"public"` or `"private"`)